store/tikv/latch/latch.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package latch
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spaolacci/murmur3"
+)
+
+// Latch stores a key's waiting transactions
+type Latch struct {
+	waiting      []uint64
+	lastCommitTs uint64
+	lock         sync.Mutex
+}
+
+// acquire tries to get current key's lock for the transaction with startTs
+// acquire is true when success
+// timeout is true when the startTs is already timeout
+// newWait is true when current transaction is new for the current latch
+func (l *Latch) acquire(startTs uint64) (acquire, timeout, newWait bool) {
+	l.lock.Lock()
+	defer l.lock.Unlock()
+	timeout = startTs <= l.lastCommitTs
```

Comment:
timeout will not consumed after this line. Why not just directly use if statement like `if startTs <= l.lastCommitTs return`? This has much more readability.